### PR TITLE
Introduce model_visitor

### PR DESCRIPTION
**Pull request by privat**
As requested, this is a basic visitor framework on models.
Services names are inspired from those in the AST visitor.

Fell free to comment the API or suggest ideas.

### CODE REVIEW

**Comment by privat** [comment]
test this please

**Comment by BlackMinou** [comment]
+1

File: src/model/model_visitor.nit [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+#   end
+# end
+# ~~~
+module model_visitor
+
+import model
+
+# The abstract model visitor template
+#
+# Specific visitor must implements the `visit` method to perform the work.
```
Comment (R4PaSs) [inline src/model/model_visitor.nit]: :s/implements/implement

**Comment by R4PaSs** [comment]
Easy enough, :shipit:

**Comment by privat** [comment]
rerolled: improve doc

**Comment by privat** [comment]
test this please

**Comment by privat** [comment]
ping @Morriar

**Pull request merged**